### PR TITLE
upgraded chef infra version from 16.13 to 17.5

### LINKED ISSUE #1923: Upgrade to Chef Infra Client 16 (and then 17)
We need to get on a supported release of Chef Infra Client. Upgrade to Chef Infra Client 16 and then on to 17.

### PR DESCRIPTION
**Pull request by RajeshPaul38**
Signed-off-by: Rajesh Paul <rajesh.paul@progress.com>

### Description

upgraded chef infra version from 16.13 to 17.5

### Issues Resolved

#1923 

### Check List

- [ ] New functionality includes tests
- [ ] All buildkite tests pass
- [x] Full omnibus build and tests in buildkite pass
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/main/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [X] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

**Comment by RajeshPaul38** [comment]
Buildkite: https://buildkite.com/chef/chef-supermarket-main-omnibus-adhoc/builds/202#a0cf2a2b-66ad-4798-aef8-24a2822a41ba

File: src/supermarket/Gemfile [hunk 1]
```diff
@@ -14,7 +14,7 @@ gem "sidekiq", "~> 4.2"
 gem "sidekiq-cron"
 
 gem "aws-sdk-s3"
-gem "chef", "~> 16.13", require: false
+gem "chef", "~> 17.4", require: false
```
Comment (tas50) [inline src/supermarket/Gemfile:17]:
```suggestion
gem "chef", "~> 17.5", require: false
```

**Comment by tas50** [comment]
We'll need to do validation that all the supermarket-ctl commands still work after this. Knife is no longer a part of chef 17 so if we use that, things will break.

**Comment by dheerajd-msys** [comment]
We need to update ruby from 2.6 to 2.7 here for making ohai compliant

**Comment by RajeshPaul38** [comment]
> We need to update ruby from 2.6 to 2.7 here for making ohai compliant

@dheerajd-msys we are already using ruby 2.7.4 as the dependency.
Is there anywhere else also that we should change?

**Comment by RajeshPaul38** [comment]
> We need to update ruby from 2.6 to 2.7 here for making ohai compliant

ok let me change in the github actions yml files. @dheerajd-msys

**Comment by RajeshPaul38** [comment]
Chef version upgraded to 17.5

Screenshot 2021-09-30 at 7 34 04 PM

**Comment by RajeshPaul38** [comment]
buildkite: https://buildkite.com/chef/chef-supermarket-main-omnibus-adhoc/builds/205#b98229c1-0b00-4ee3-b324-994de625abb5

**Comment by RajeshPaul38** [comment]
> We'll need to do validation that all the supermarket-ctl commands still work after this. Knife is no longer a part of chef 17 so if we use that, things will break.

@tas50 I've checked all the ctl commands. Thore are working fine.

File: omnibus/config/projects/supermarket.rb [hunk 2]
```diff
@@ -47,6 +48,7 @@
 dependency "version-manifest"
 
 # remove lots of ruby clutter we don't need
```
Comment (dheerajd-msys) [inline omnibus/config/projects/supermarket.rb:50]:
```suggestion
dependency "chef"
# remove lots of ruby clutter we don't need
```

**Comment by dheerajd-msys** [comment]
All ok, comment adjust and we are done !!

**Review by dheerajd-msys — APPROVED** [review]

**Pull request merged**